### PR TITLE
[Feature]: flatten nested tensordicts

### PR DESCRIPTION
**Pull request by vmoens**
## Description

Introduces new `flatten_key` function and method.

## Motivation and Context

Allows to get a single, non-nested tensordict from a nested one.

## Types of changes

New function in `tensordict.py` and method in `_TensorDict` parent class.
Tests the function over all tensordict subtypes.

### CODE REVIEW

File: torchrl/data/tensordict/tensordict.py [hunk 1]
```diff
@@ -3812,3 +3971,56 @@ def _check_keys(
...
+        # tensordict
+        out = TensorDict(
+            {},
+            [*parent_batch_size, *tensor.shape[self_batch_dims:]],
+            device=self_device,
+        )
+        return out
+
+
+def flatten_keys(
```
Comment (shagunsodhani) [inline torchrl/data/tensordict/tensordict.py:3997]: we may eventually want to have a `unflatten_key` method as well

File: torchrl/data/tensordict/tensordict.py [hunk 2]
```diff
@@ -1231,6 +1302,9 @@ def __iter__(self) -> Generator:
         for i in range(length):
             yield self[i]
 
+    def flatten_keys(self, separator=",", inplace: bool = True) -> _TensorDict:
+        return flatten_keys(self, separator=separator, inplace=inplace)
```
Comment (shagunsodhani) [inline torchrl/data/tensordict/tensordict.py:1306]: Maybe its better to mark `flatten_keys` as a static method. The current implementation (while correct) looks a bit confusing

**Review by shagunsodhani — APPROVED** [review]

File: torchrl/data/tensordict/tensordict.py (hunk 2 above)
Comment (vmoens) [reply torchrl/data/tensordict/tensordict.py:1306]: ok i removed the function and made it a method. 
not sure we can make it static as we need the `self` arg

File: torchrl/data/tensordict/tensordict.py (hunk 1 above)
Comment (vmoens) [reply torchrl/data/tensordict/tensordict.py]: makes sense I can do that

Comment (vmoens) [reply torchrl/data/tensordict/tensordict.py]: See latest commit :)

**Pull request merged**